src/compiler/checker.ts

```diff
@@ -34287,10 +34291,69 @@ namespace ts {
                     return isLiteralImportTypeNode(node) ? getSymbolAtLocation(node.argument.literal) : undefined;
 
                 case SyntaxKind.ExportKeyword:
-                    return isExportAssignment(node.parent) ? Debug.assertDefined(node.parent.symbol) : undefined;
+                    if (isExportAssignment(node.parent)) {
+                        return Debug.assertDefined(node.parent.symbol);
+                    }
+                    break;
+            }
+        }
 
-                default:
-                    return undefined;
+        /**
+         * Gets the symbol related to the provided location, if it that location is a keyword.
+         * These additional keywords are normally only used to resolve references but would
+         * not be used for document highlights, quickinfo, etc.
+         */
+        function getSymbolAtKeyword(node: Node): Symbol | undefined {
+            if (node.flags & NodeFlags.InWithStatement) {
+                // We cannot answer semantic questions within a with block, do not proceed any further
+                return undefined;
+            }
+
+            const { parent } = node;
+
+            // If the node is a modifier of its parent, get the symbol for the parent.
+            if (isModifier(node) && contains(parent.modifiers, node)) {
+                return getSymbolOfNode(parent);
+            }
+
+            switch (node.kind) {
+                case SyntaxKind.InterfaceKeyword:

```

Comment:
Class and function ?

Reply:
Why is class and function different from interface, namespace, enum etc  in terms of supported functionality? (its not like the behavior is only for class or function without name)

Reply:
Class and function were previously supported in a very special way to handle unnamed class and function expressions. All other declaration types are guaranteed to have names and are thus not specialized in this way.

Reply:
Yeah but what I am saying is that behavior is not just for unnamed class … It happens for named class too.. eg. if you click on class the document highlights the references of className instead.. So why do interface and other such keywords that are declarations need to have different behaviour

Reply:
Per https://github.com/microsoft/TypeScript/issues/28268#issuecomment-577941659, we don't actually want that behavior for other elements (and probably don't want it for `class` and `function` either).

Reply:
Per some of your other comments, I'm going to move this to services and leave the public API for `getSymbolAtLocation` as is.